backend/src/v4/handler/elastic.js

```diff
@@ -94,10 +94,14 @@ const establishIndices = async (client)=>{
 
 const elasticClientPromise = createElasticClient();
 
-const createElasticRecord = async (index, body, id) => {
+Elastic.createElasticRecord = async (index, body, id) => {
 	try {
 		const elasticClient = await elasticClientPromise;
 		if (elasticClient && body) {
+			const { namespace } = elasticConfig;
+			if (namespace) {
+				body.namespace = namespace;
```

Comment:
do we need to do anything to `establishIndices`?

We seem to do that to ensure the indices are created, but that object doesnt' have the additional `namespace` value.

We also don't do this for the io-activity stuff...

Reply:
also can we call it `Namespace` since everything seems to begin with a capital letter (other than license...)

Reply:
Can we also add the hostname by default too? Instead of just adding it to io-activity.

Reply:
On the existing `prod` `io-activity` all field names are lowercase.

Index mapping for `io-activity` added. By default, it will create the mapping as data is added, but by enforcing a mapping we can "define how each field should be treated by Elasticsearch, whether it should be tokenized and filtered, or whether it should be a specific type like integer, date, etc."

`host` added by default on record creation if it exists in `config`.

![image](https://github.com/3drepo/3drepo.io/assets/1438332/58e819dd-3c67-4c3d-bea2-07d679aa8e54)

Reply:
since w'ere adding host and namespace to login activity as well can we update the ensure index fit to match?